suave/backends/redis_pubsub.go

```diff
@@ -0,0 +1,177 @@
+package backends
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/flashbots/go-utils/cli"
+	"github.com/go-redis/redis/v8"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	suave "github.com/ethereum/go-ethereum/suave/core"
+)
+
+var (
+	redisUpsertTopic = "store:upsert"
+
+	redisConnectionPoolSize = cli.GetEnvInt("REDIS_CONNECTION_POOL_SIZE", 0) // 0 means use default (10 per CPU)
+	redisMinIdleConnections = cli.GetEnvInt("REDIS_MIN_IDLE_CONNECTIONS", 0) // 0 means use default
+	redisReadTimeoutSec     = cli.GetEnvInt("REDIS_READ_TIMEOUT_SEC", 0)     // 0 means use default (3 sec)
+	redisPoolTimeoutSec     = cli.GetEnvInt("REDIS_POOL_TIMEOUT_SEC", 0)     // 0 means use default (ReadTimeout + 1 sec)
+	redisWriteTimeoutSec    = cli.GetEnvInt("REDIS_WRITE_TIMEOUT_SEC", 0)    // 0 means use default (3 seconds)
+
+	formatRedisBidKey = func(bidId suave.BidId) string {
+		return fmt.Sprintf("bid-%x", bidId)
+	}
+
+	formatRedisBidValueKey = func(bidId suave.BidId, key string) string {
+		return fmt.Sprintf("bid-data-%x-%s", bidId, key) // TODO: should also include the hash of the bid at least
+	}
+)
+
+type RedisPubSub struct {
```

Comment:
I think you also need to include a UUID from the node. Last time I worked with redis pubsub, it would pub it's own message back to itself and you are meant to filter these out with the UUID.

Reply:
I was thinking to maybe set it to the execution node address, but that would mean nodes can't share it. Uuid is easily spoofed, but we could use both as a way to cut out the messages we send

Reply:
Check if you like it better